### ISSUE TITLE
Add support for Goiardi's bookshelf implementation

### ISSUE THREAD

**Issue by oker1** [schubergphilis/chef-guard#79]
I've tried to freeze a cookbook but I get an error:

```
ERROR:   2015/02/10 10:22:56 Failed to get organization ID for : Could not find an organization ID in reply: {"checksums":{"00000000000000000000000000000000":{"needs_upload":true,"url":"http://127.0.0.1:8889/file_store/00000000000000000000000000000000"}},"sandbox_id":"24a1e661d3034f2f5c8149198b5e0d14","uri":"http://127.0.0.1:8889/sandboxes/24a1e661d3034f2f5c8149198b5e0d14"}
```

Seems like the cookbook module assumes enterprise chef or chef 12 since opensource chef 11 does not have organizations.

**Comment by svanharmelen** [comment]
@oker1 there are a few companies using CG with open source Chef 11, so this should work just fine. Which version of CG are you running? And did you provide the correct values for:

```
[chef]
  s3key           = xxx
  s3secret        = xxx
```

Those values can normally be found in `/etc/chef-server/chef-server-secrets.json` on an open source Chef 11 server.

**Comment by oker1** [comment]
To be precise I'm using goiardi. It doesn't use these secrets, AFAIK the opscode chef server uses these only if the bookshelf uses s3 as storage, so it shouldn't be needed. The cookbook upload without freezing works. I'm running the current master HEAD.

I think the problem is around [here](https://github.com/xanzy/chef-guard/blob/c0f4adfc2054eae8aa543468f35b8e54d1bc7133/cookbook.go#L106).

**Comment by svanharmelen** [comment]
No, that is not correct... The bookshelf always uses these keys to authenticate calls to the bookshelf, so even when you are using the default (native Chef) bookshelf backend you get with a standard install.

The fact that a cookbook upload without freezing works makes a lot of sense as there isn't done much checking when you don't freeze (see [here](http://xanzy.io/projects/chef-guard/workflows/cookbook_upload.html) for more details).

Like mentioned before, there are people running CG with open source Chef 11 without any issues (tested and confirmed) so it seems like this is a slight incompatibility between Chef and Goiardi where Goiardi clearly handles this part differently than an actual Chef server.

After having a quick look at the code of Goiardi I do indeed notice there isn't a `bookshelf` API endpoint at all, so that pretty much explains this problem. I see how Goiardi handles files and it shouldn't be to difficult to support this in CG, so will label this as enhancement and will try to get support in before making the next release (which I hope to do this week).

**Comment by oker1** [comment]
Great, thanks for looking into it!

**Comment by svanharmelen** [comment]
Fixed by latest Chef-Guard release (v0.6.0 PR #97)
